pkg/node/volumemgr.go

```diff
@@ -886,56 +889,60 @@ func (m *VolumeManager) discoverLVGOnSystemDrive() error {
 		}
 	}
 
-	var (
-		rootMountPoint, vgName string
-		vgFreeSpace            int64
-	)
-
-	if rootMountPoint, err = m.fsOps.FindMountPoint(base.KubeletRootPath); err != nil {
-		ll.Errorf("Failed to find root mountpoint for %s, error: %v, try to find for %s",
-			base.KubeletRootPath, err, base.HostRootPath)
-		if rootMountPoint, err = m.fsOps.FindMountPoint(base.HostRootPath); err != nil {
-			return fmt.Errorf(errTmpl, err)
-		}
+	// 2. check whether there is LVG configuration on the system drive or not
+	var driveCR = new(drivecrd.Drive)
+	// TODO: handle situation when there is more then one system drive
+	if err = m.k8sClient.ReadCR(context.Background(), m.systemDrivesUUIDs[0], driveCR); err != nil {
+		return err
 	}
 
-	// from container we expect here name like "VG_NAME[/var/lib/kubelet/pods]"
-	rootMountPoint = strings.Split(rootMountPoint, "[")[0]
-
-	devices, err := m.listBlk.GetBlockDevices(rootMountPoint)
+	pvs, err := m.lvmOps.GetAllPVs()
 	if err != nil {
-		return fmt.Errorf(errTmpl, err)
+		return fmt.Errorf("unable to list PVs on the system: %v", err)
 	}
 
-	if devices[0].Rota != base.NonRotationalNum {
+	var systemPVName string
+	for _, pv := range pvs {
+		if strings.Contains(pv, driveCR.Spec.Path) {
```

Comment:
I think we should avoid using Contains here. If system has many drives we can hit error with collision between /dev/sda and /dev/sdaa

Reply:
Yes, I agreed, but we can't use `==` here because of PV might be created on some partition, e.g. `/dev/sda1`. I will handle case that you described.

Reply:
I think we can rely on partition name like `DEVICE_NAME` + `NUBMER`
https://www.debian.org/releases/buster/amd64/apcs04.en.html#:~:text=Linux%20represents%20the%20primary%20partitions,drive%20is%20%2Fdev%2Fsda5%20.

Reply:
refactored

Reply:
Maybe we can remove Contains check and use single regexp?
```regexp.Match(fmt.Sprintf("^%s\\d*$", a), driveCR.Spec.Path)```

Reply:
yeah, done